src/main/java/org/opensearch/ad/task/ADBatchTaskRunner.java

```diff
@@ -413,81 +486,84 @@ public void forwardOrExecuteADTask(
-                        adTaskManager.createADTaskDirectly(adEntityTask, r -> {
-                            adEntityTask.setTaskId(r.getId());
+                .getAndExecuteOnLatestADTask(
+                    detectorId,
+                    entityString,
+                    ImmutableList.of(ADTaskType.HISTORICAL_HC_ENTITY),
+                    existingEntityTask -> {
+                        if (existingEntityTask.isPresent()) { // retry failed entity caused by limit exceed exception
+                            // TODO: if task failed due to limit exceed exception in half way, resume from the break point or just clear the
+                            // old AD tasks and rerun it? Currently we just support rerunning task failed due to limit exceed exception
+                            // before starting.
+                            ADTask adEntityTask = existingEntityTask.get();
+                            logger
+                                .debug(
+                                    "Rerun entity task for task id: {}, error of last run: {}",
+                                    adEntityTask.getTaskId(),
+                                    adEntityTask.getError()
+                                );
                             ActionListener<ADBatchAnomalyResultResponse> workerNodeResponseListener = workerNodeResponseListener(
                                 adEntityTask,
                                 transportService,
                                 listener
                             );
                             forwardOrExecuteEntityTask(adEntityTask, transportService, workerNodeResponseListener);
-                        }, wrappedListener);
-                    }
-                }, transportService, false, wrappedListener);
+                        } else {
+                            logger.info("Create entity task for entity:{}", entityString);
+                            Instant now = Instant.now();
+                            String parentTaskId = adTask.getTaskType().equals(ADTaskType.HISTORICAL_HC_ENTITY.name())
+                                ? adTask.getParentTaskId()
+                                : adTask.getTaskId();
+                            Entity entity = adTaskManager.parseEntityFromString(entityString, adTask);
+                            ADTask adEntityTask = new ADTask.Builder()
+                                .detectorId(adTask.getDetectorId())
+                                .detector(detector)
+                                .isLatest(true)
+                                .taskType(ADTaskType.HISTORICAL_HC_ENTITY.name())
+                                .executionStartTime(now)
+                                .taskProgress(0.0f)
+                                .initProgress(0.0f)
+                                .state(ADTaskState.INIT.name()) // TODO where to set INIT state
+                                .initProgress(0.0f) // TODO where to set INIT state
```

Comment:
nit: unsure what these comments mean, can they be removed?

Reply:
sure, will remove these comments